### ISSUE TITLE
get/query methods failing silently

### ISSUE THREAD

**Issue by squared99** [googleads/googleads-php-lib#183]
For some reason any call to 'get' or 'query' appears to be failing for me. No exception or SoapFault is thrown, the code just halts and appears to be failing silently.

I've successfully made calls to 'mutate' and creating the AdWordsUser using the same initialization and credentials with no issues, and so have been able to create and update campaigns, user lists, ad groups, and budgets.

I've been able to trace the issue with 'get' as far as line 215 in AdsSoapClient.php.
`$response = parent::__soapCall($function_name, $arguments, $options,
          $input_headers, $output_headers);`

This call is where it fails and halts. No code will run after this point.
 
So basically the code sample for BasicOperations/AddCampaigns.php works just fine, but BasicOperations/GetCampaigns.php does not.  It fails on $page = $campaignService->get($selector);

Im running PHP 5.6 on Ubuntu precise (12.04.5 LTS).

Hoping for any insight as I cannot find any way to debug this issue. No exceptions are thrown, nothing appears in the logs for php or the library soap logs.

Many Thanks

**Comment by fiboknacky** [comment]
Hi @squared99 

Could you please answer some questions below?
- What version of client library do you use?
- How did you set up this client library? By composer? Do you try running it as is or plugged into other source codes?
- Have you modified any codes in this library?
- It failed silently or it took so _long_ time (e.g., 15 mins) to finish? I asked because I've found the case in the past when there is some connection/networking problem, it took so long time to finish.

Thanks in advance.

Best,
Knack

**Comment by squared99** [comment]
Hi There,

Thanks very much for follow-up. Answers to questions below:

What version of client library do you use?
I’m using v201609, but have also swapped to v201607 to test and we see the same results.

How did you set up this client library? By composer? Do you try running it as is or plugged into other source codes?
Yes, it was installed via composer. We include the libraries in our code via paths to our init.php file which defines the paths to the vendor folder.

Have you modified any codes in this library?
No, the files installed via composer are unchanged.

It failed silently or it took so long time (e.g., 15 mins) to finish? I asked because I've found the case in the past when there is some connection/networking problem, it took so long time to finish.
It fails pretty immediately. As part of the code where this issue became apparent there are actually prior calls to initialize the user and modify an existing campaign which work fine and we will see output in the soap logs, but when it hits the ‘get’ call with the selector defined it will fail.

Other points:

If I don’t send the selector or send an invalidly formatted selector to the ‘get’ call we will get an immediate and properly formatted soap fault response about invalid parameters, and an entry will be recorded in the logs. So it seems like it is an issue with processing a valid response in some way. 

We’ve tried ‘get’ calls on differing service objects and they all fail. We’ve limited the amount of pages to return to 1 with no change in behavior.

Best

**Comment by hakimio** [comment]
@squared99 Did you by any chance chance enable wsdl cache?

**Comment by fiboknacky** [comment]
Hello @squared99,

I've never seen this issue before and what @hakimio kindly pointed out looks feasible to me.
Could you please confirm on this point too?
I would also try using the client library on other computers / network settings as well.

Best,
Knack

**Comment by vtsao** [comment]
Closing due to lack of activity, please re-open if this is still an issue.
